image_view/src/nodes/video_recorder.cpp

```diff
@@ -70,9 +71,14 @@ void callback(const sensor_msgs::ImageConstPtr& image_msg)
 
     }
 
-    if ((image_msg->header.stamp - g_last_wrote_time) < ros::Duration(1.0 / fps))
+    if ((image_msg->header.stamp - g_last_wrote_time) < ros::Duration((1.0 / fps)*(1.0-tolerance/100)))
```

Comment:
This still doesn't sovle the case where the frame arrives early, only when it arrives late within the tolerance value. I recommend using `std::abs()` to handle both cases.

Reply:
I'm not sure I understand your concern. My new condition was specifically designed to handle the case where the frame arrives early. In the original code, the frame-skipping only happens when frames arrive too early (when `image_msg->header.stamp - g_last_wrote_time` is LESS THAN 1 frame duration). If frames are late, they are not skipped.

My proposed change is to only skip frames if when they come WAY earlier than they should. This new conditions checks if the new frame arrived earlier than `1.0-tolerance` times the frame duration, which IS correct. 

The result: late frames are not skipped, and frames earlier than the frame duration-tolerance are skipped.

Am I misunderstanding something here?

Reply:
Nope, you didn't miss anything - I did. You're correct. Thanks for the clarification.

Reply:
It seems like you might be confusing durations with rates here. When a frame comes early, it has a SHORTER duration since the last frame was saved. Thus the tolerance condition should check if the frame arrival duration is smaller than the frame duration MINUS some tolerance. If so, then the frame should be skipped, otherwise, the frame should not be skipped (including if it arrives late, right?).